packages/truffle-core/lib/mnemonics/mnemonic.js

```diff
@@ -0,0 +1,69 @@
+/**
+* @module mnemonic;
+* @requires module:truffle-config
+* @requires module:seedrandom
+* @requires module:bip39
+* @requires module:ethereumjs-wallet/hdkey
+* @requires module:crypto
+*/
+
+const Config = require('truffle-config');
+const defaultUserConfig = Config.getUserConfig();
+const seedrandom = require('seedrandom');
+const bip39 = require('bip39'); 
+const hdkey = require('ethereumjs-wallet/hdkey');
+const crypto = require('crypto');
+
+const mnemonic = {
+
+  /**
+  * gets user-level mnemonic from user config, and if missing generates a new mnemonic
+  * @returns {String} mnemonic
+  */
+  getOrGenerateMnemonic: function () {
+    let mnemonic;
+    const userMnemonicExists = defaultUserConfig.get("mnemonic");
+    if(!userMnemonicExists) {
+      mnemonic = bip39.entropyToMnemonic(crypto.randomBytes(16).toString('hex'));
+      defaultUserConfig.set({"mnemonic": mnemonic});
+    } else {
+      mnemonic = defaultUserConfig.get("mnemonic");
+    }
+
+    return mnemonic;
+  },
+
+  /**
+  * gets accounts object using mnemonic 
+  * @param {String}   
+  * @returns {Object} mnemonicObject
+  */
+  getAccountsInfo: function () {
+
+    let mnemonic = this.getOrGenerateMnemonic();
+    let accounts = [];
+    let privateKeys = [];
+    let mnemonicObject = {};
```

Comment:
You can remove this declaration I think.

Reply:
oh yeah, will do, thanks!